member.php

```diff
@@ -1599,8 +1600,29 @@
 			$password_length = 8;
 		}
 
-		$password = random_str($password_length);
-		$logindetails = update_password($user['uid'], md5($password), $user['salt']);
+		// Set up user handler.
+		require_once 'inc/datahandlers/user.php';
+		$userhandler = new UserDataHandler('update');
+
+		$userhandler->set_data(array(
+			'uid'		=> $user['uid'],
+			'username'	=> $user['username'],
+			'email'		=> $user['email'],
+			'password'	=> random_str($password_length)
+		));
+
+		if(!$userhandler->validate_user())
+		{
+			error($lang->error_unknownerror);

```

Comment:
Yes, but I mean that the friendly errors and `false` are always returned in pairs: https://github.com/mybb/mybb/blob/feature/inc/datahandlers/user.php#L193-194

So if `verify_password()` returns false, `$errors[0]` will be always set and `error($errors[0]);` executed, thus `error($lang->error_unknownerror);` will be never used.